Tests/SparkConnectTests/DataFrameWriterTests.swift

```diff
@@ -0,0 +1,117 @@
+struct DataFrameWriterTests {
+
+  @Test
+  func csv() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.csv(tmpDir)
+    #expect(try await spark.read.csv(tmpDir).count() == 2025)
+    await spark.stop()
+  }
+
+  @Test
+  func json() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.json(tmpDir)
+    #expect(try await spark.read.json(tmpDir).count() == 2025)
+    await spark.stop()
+  }
+
+  @Test
+  func orc() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.orc(tmpDir)
+    #expect(try await spark.read.orc(tmpDir).count() == 2025)
+    await spark.stop()
+  }
+
+  @Test
+  func parquet() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.parquet(tmpDir)
+    #expect(try await spark.read.parquet(tmpDir).count() == 2025)
+    await spark.stop()
+  }
+
+  @Test
+  func pathAlreadyExist() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.csv(tmpDir)
+    try await #require(throws: Error.self) {
+      try await spark.range(2025).write.csv(tmpDir)
+    }
+    await spark.stop()
+  }
+
+  @Test
+  func overwrite() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.range(2025).write.csv(tmpDir)
+    try await spark.range(2025).write.mode("overwrite").csv(tmpDir)
+    await spark.stop()
+  }
+
+  @Test
+  func save() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    for format in ["csv", "json", "orc", "parquet"] {
+      try await spark.range(2025).write.format(format).mode("overwrite").save(tmpDir)
+      #expect(try await spark.read.format(format).load(tmpDir).count() == 2025)
+    }
+    await spark.stop()
+  }
+
+  @Test
+  func partitionBy() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    try await spark.sql("SELECT 1 col1, 2 col2").write.partitionBy("col2").csv(tmpDir)
+    #expect(try await spark.read.csv("\(tmpDir)/col2=2").count() == 1)
+    await spark.stop()
+  }
+
+  @Test
+  func sortByBucketBy() async throws {
+    let tmpDir = "/tmp/" + UUID().uuidString
+    let spark = try await SparkSession.builder.getOrCreate()
+    let df = try await spark.sql("SELECT 1 col1, 2 col2")
+    try await #require(throws: Error.self) {
+      try await df.write.sortBy("col2").csv(tmpDir)
+    }
+    try await #require(throws: Error.self) {
```

Comment:
Nit: technically, Spark produces 3 different errors
- when only `sortBy` and
- when only `bucketedBy` and 
- when both

are specified, so adding a `bucketBy` only test case might make sense.